src/app/shared/auth/store/actions.ts

```diff
@@ -1,33 +1,35 @@
-import { action, actionWithPayload } from '@store/action-factory';
+import { createAction } from '@reduxjs/toolkit';
```

Comment:
In order to have typed actions payload and work fine with `redux-observable` let's create the following utility function in `src/app/shared/store/utils/define-action.ts`:

```ts
import { ActionCreatorWithoutPayload, ActionCreatorWithPayload, createAction } from '@reduxjs/toolkit';

export function defineAction<T extends object | void = void>(type: string): T extends object ? ActionCreatorWithPayload<T> : ActionCreatorWithoutPayload {
  return createAction(type) as any;
}
```

And will use it instead of `createAction`:

```ts
import { defineAction } from '@store';

const someActionWithPayload = defineAction<{ page?: number }>(
  '[Some Feature] Some action'
);
```